src/main/java/com/tabnine/intellij/completions/InlayHoverMouseMotionListener.java

```diff
@@ -0,0 +1,155 @@
+package com.tabnine.intellij.completions;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.Inlay;
+import com.intellij.openapi.editor.colors.EditorColorsManager;
+import com.intellij.openapi.editor.event.EditorMouseEvent;
+import com.intellij.openapi.editor.event.EditorMouseMotionListener;
+import com.intellij.openapi.ui.popup.Balloon;
+import com.intellij.openapi.ui.popup.JBPopupFactory;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.ui.JBColor;
+import com.intellij.ui.awt.RelativePoint;
+import com.intellij.util.DocumentUtil;
+import com.tabnine.binary.requests.notifications.HoverBinaryResponse;
+import com.tabnine.binary.requests.notifications.actions.HoverActionRequest;
+import com.tabnine.general.DependencyContainer;
+import com.tabnine.general.NotificationOption;
+import com.tabnine.general.StaticConfig;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.event.HyperlinkListener;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class InlayHoverMouseMotionListener implements EditorMouseMotionListener {
+    //URLs for adding the tabnine icon to the inlay popup.
+    private final static String ICON_AND_NAME_URL;
+    private final static String ICON_AND_NAME_DARK_URL;
+
+    private final HoverBinaryResponse hoverBinaryResponse;
+    private final Inlay inlay;
+    private Balloon balloon;
+
+    //This is a hack to get the light/dark icon URL for use in the popup html.
+    static {
+        ICON_AND_NAME_URL = "jar:" + LimitExceededLookupElement.class.getClassLoader()
+                .getResource(StaticConfig.ICON_AND_NAME_PATH).getPath();
```

Comment:
Are we sure there is no more idiomatic way of doing this? Seems like bringing a gun to a knive fight

Reply:
I didn't find any. There doesn't seem to be any API for getting the icon's URL.